swagger_server/controllers/neighbours_put_controller.py

```diff
@@ -0,0 +1,36 @@
+import connexion
+
+from swagger_server import registry
+from swagger_server.exceptions import NotFound
+from swagger_server.factories import GraphFactory
+from swagger_server.models import Sample
+from swagger_server.models.error import Error  # noqa: E501
+from swagger_server.models.neighbour import Neighbour  # noqa: E501
+from swagger_server.repositories import Neo4jRepository
+
+
+def samples_id_nearest_neighbours_put(body, id):  # noqa: E501
+    """samples_id_nearest_neighbours_put
+
+    Replace the list of nearest neighbours of a sample based on a sample ID. # noqa: E501
+
+    :param body: New list of nearest neighbours to replace old one.
+    :type body: list | bytes
+    :param id:
+    :type id: str
+
+    :rtype: List[Neighbour]
+    """
+    if connexion.request.is_json:
+        body = [Neighbour.from_dict(d) for d in connexion.request.get_json()]  # noqa: E501
+
+    repo = registry.get('repo')
+
+    try:
+        sample = Sample(id, body)
+        node = GraphFactory.build(sample)
+        repo.update(node)
```

Comment:
I think this method should returns 400 if the input contains nearest-leaf, what do you think?

Reply:
What I was asking is what will happen in this scenario: 

Image that there are three samples SA, SB, SC and one tree leaf LA already in graph. The sample SA is connected to a tree leaf LA. The sample SA is connected to both SB and SC.

Now if we want to call this method with SA and new neighbours list [SC, SD]. My expectation for the results is `200` response code and [SC] in the response body. As a result of this update, the graph has SA, SB, SC, LA. SA is connected to SC and LA.

Reply:
Since this endpoint's purpose is to update the neighbours, I suggest it does not touch the existing leaf relationship SA-LA, either by ignoring the leaf info in the input (if exists), or explicitly tell the client side to remove the leaf info from the input first.

Reply:
Yes, that endpoint is only about creating new or replacing the old neighbouring relationship for a given sample. The consequences of requests to this endpoint should not touch the leaf relationship in any way. There is another endpoint to deal with the leaf relationship.